SpellCheck/src/test/java/AIR/SpellCheck/Engines/HunspellEngineTest.java

```diff
@@ -0,0 +1,77 @@
+package AIR.SpellCheck.Engines;
+
+import AIR.SpellCheck.SpellEngineFactory;
+import dk.dren.hunspell.Hunspell;
+import org.assertj.core.util.Lists;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HunspellEngineTest {
+    private final static String LANGUAGE_CODE = "languageCode";
+    private final static String WORD = "word";
+
+    @Mock
+    private SpellEngineFactory spellEngineFactory;
+
+    @Mock
+    private Hunspell.Dictionary dictionary;
+
+    private HunspellEngine engine;
+
+    @Before
+    public void setup() {
+        engine = spy(new HunspellEngine(spellEngineFactory));
```

Comment:
Nice for wrapping and intercepting static method calls.  Still feels ugly, though.  The interface *should* have been designed with a language parameter rather than requiring a reach-back to manually parse a request param... But, one thing at a time.